inngest/src/main/kotlin/com/inngest/State.kt

```diff
@@ -20,6 +23,19 @@ class State(
         return sb.toString()
     }
 
+    private fun findNextAvailableStepId(id: String): String {
+        if (id !in stepIdsToSeenCount) {
+            stepIdsToSeenCount[id] = 1
+            return id
+        }
+
+        // use the seen count so far for current step, increment for next time
+        val stepNumber = stepIdsToSeenCount[id]
+        stepIdsToSeenCount[id] = stepIdsToSeenCount.getValue(id) + 1
```

Comment:
`stepIdsToSeenCount[id]++` does not compile because the return value on `[]` map access is `T?`, even with the `!in` check above 😭 

https://blog.danlew.net/2017/06/14/convincing-the-kotlin-compiler-that-code-is-safe/